R/zzz.R

```diff
@@ -289,13 +289,17 @@ settings <- NULL
   base_ns <- getNamespace("base")
   backports_ns <- getNamespace("backports")
   lintr_ns <- getNamespace(pkgname)
+  # NB: we're _probably_ meant to use utils::assignInNamespace() here, but
+  #   there's a rather clunky R CMD check under tools:::.check_package_code_tampers()
+  #   that requries such calls to pass asNamespace("lintr") explicitly instead of
+  #   re-using the getNamespace() outcome as we do here.
   for (base_fun in c("str2lang", "str2expression")) {
     if (!exists(base_fun, base_ns)) {
       assign(base_fun, get(base_fun, backports_ns), lintr_ns)
     }
   }
 
-  default_settings <<- list(
+  assign("default_settings", envir = lintr_ns, list(
```

Comment:
Use `assignInMyNamespace()`?

Reply:
see the comment... I tried that but it threw an R CMD check warning (it looks for potentially risky calls to assignInNamespace(), flagging anything that doesn't use asNamespace("lintr") exactly). we could match that, but we already use assign() like this for backports, so I continued to do so.

Reply:
There is assignIn*My*Namespace for that particular use case, no?

Reply:
> There is assignIn_My_Namespace for that particular use case, no?

Ah, sorry, read on phone. Trying now.

Side note/related: https://bugs.r-project.org/show_bug.cgi?id=18595